src/lib/util/options.cpp

```diff
@@ -462,6 +465,42 @@ bool core_options::parse_ini_file(util::core_file &inifile, int priority, int ig
 
 
 //-------------------------------------------------
+//	find_within_command_line - finds a specific
+//	value from within a command line
+//-------------------------------------------------
+
+const char *core_options::find_within_command_line(int argc, char **argv, const char *optionname)
+{
+	// find this entry within the options (it is illegal to call this with a non-existant option
+	// so we assert if not present)
+	auto curentry = m_entrymap.find(optionname);
+	assert(curentry != m_entrymap.end());
+
+	// build a vector with potential targets
+	std::vector<std::string> targets;
+	const char *potential_target;
+	int index = 0;
+	while ((potential_target = curentry->second->name(index++)) != nullptr)
+	{
+		// not supporting unadorned options for now
+		targets.push_back(std::string("-") + potential_target);
+	}
+
+	// find each of the targets in the argv array
+	for (int i = 1; i < argc - 1; i++)
+	{
+		auto iter = std::find_if(targets.begin(), targets.end(), [argv, i](const std::string &targ)
```

Comment:
Could you make the indentation here more obvious?  Also, `const`!  Like

    auto const iter = std::find_if(
            targets.begin(),
            targets.end(),
            [argv, i] (const std::string &targ) { return targ == argv[i]; });